functions/concepts/TimeBoundedRetries/Function.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_tips_infinite_retries]
+using CloudNative.CloudEvents;
+using Google.Cloud.Functions.Framework;
+using Google.Events.SystemTextJson.Cloud.PubSub.V1;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TimeBoundedRetries
+{
+    public class Function : ICloudEventFunction<MessagePublishedData>
+    {
+        private static readonly TimeSpan MaxEventAge = TimeSpan.FromSeconds(10);
+        private readonly ILogger _logger;
+
+        // Note: for additional testability, use an injectable clock abstraction.
+        public Function(ILogger<Function> logger) =>
+            _logger = logger;
+
+        public Task HandleAsync(CloudEvent cloudEvent, MessagePublishedData data, CancellationToken cancellationToken)
+        {
+            string textData = data.Message.TextData;
+
+            DateTime utcNow = DateTime.UtcNow;
+
+            // Every PubSub CloudEvent will contain a timestamp.
+            DateTime timestamp = cloudEvent.Time.Value.ToUniversalTime();
+            DateTime expiry = timestamp + MaxEventAge;
+
+            // Ignore events that are too old.
+            if (utcNow > expiry)
+            {
+                _logger.LogInformation("Dropping PubSub message '{text}'", textData);
+                return Task.CompletedTask;
+            }
+
+            // Process events that are recent enough.
+            // If this processing throws an exception, the message will be retried until either
```

Comment:
Ahh, I think this answers my question on the previous sample.
OK, then what is confusing about the previous sample is the use of the retry JSON property. Maybe send a fail JSON property instead, and add this same comment there or similar:
"When processing a PubSub message, if the function throws, the message will be retried".

Reply:
I'd prefer to keep it consistent with the other languages' samples, although I agree that would probably be clearer. (Or maybe put two numbers to divide in the message, so that it fails if the divisor is zero...)